CopilotKit/packages/backend/src/lib/openai-adapter.ts

```diff
@@ -40,6 +40,14 @@ export class OpenAIAdapter implements CopilotKitServiceAdapter {
       maxTokensForOpenAIModel(forwardedProps.model || this.model),
     );
 
+    // remove message.function_call.scope if it's present.
+    // scope is a field we inject as a temporary workaround (see elsewhere), which openai doesn't understand
+    messages.forEach((message) => {
+      if (message.function_call?.scope) {
+        delete message.function_call.scope;
+      }
+    });
+
```

Comment:
the presence of `function_call?.scope` causes an error in openai calls with the latest GPT-4-Turbo model.

So we remove it here before calling openai.

@mme: can you please verify that the order of operations allows for this?
We don't need this information from the return value of this function, right? (because we have it elsewhere)